src/traces/bar/plot.js

```diff
@@ -428,7 +428,7 @@ function appendBarText(gd, plotinfo, bar, cd, i, x0, x1, y0, y1, opts, makeOnCom
     }
 
     transform.fontSize = font.size;
-    recordMinTextSize(trace.type, transform, fullLayout);
+    recordMinTextSize(trace.type === 'histogram' ? 'bar' : trace.type, transform, fullLayout);
```

Comment:
Can you explain how this relates to the `clearMinTextSize` call above, where it's just `'bar'`, not `trace.type` at all?

https://github.com/plotly/plotly.js/blob/c8f9f73b5e7b44f9a5dd25c7013daafdd357e78f/src/traces/bar/plot.js#L91-L92

Reply:
Histogram plots in bar layer. But waterfall and funnel have their own layers.